File: 219.Contains-Duplicate-II.py
# https://leetcode.com/problems/contains-duplicate-ii/
#
# algorithms
# Easy (34.35%)
# Total Accepted:    176,832
# Total Submissions: 514,775
# beats 79.72% of python submissions

class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        hash_map = {}
        
        for idx, n in enumerate(nums):
            if n in hash_map:
                if idx - hash_map[n] <= k:
                    return True
            hash_map[n] = idx
            
        return False
        

File: 219.Contains-Duplicate-II.java
// https://leetcode.com/problems/contains-duplicate-ii/
//
// algorithms
// Easy (36.55%)
// Total Accepted: 234,075
// Total Submissions: 640,395


class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int len = nums.length;

        for (int i = 0; i < len; i++) {
            if (map.containsKey(nums[i]) && i - map.get(nums[i]) <= k) {
                return true;
            }
            map.put(nums[i], i);
        }

        return false;
    }
}